### PR TITLE
Feature/subspace

### PR DESCRIPTION
**Pull request by pardeep-singh**
- Add utility functions for Subspace wrapper functions.
- Add tests for utility function.
- Update README.md with Subspace utility functions example.

### CODE REVIEW

File: src/clj_fdb/core.clj [hunk 1]
```diff
@@ -120,3 +123,110 @@
...
+  - TransactionContext `tc`
+  - Subspace `s` which will be used to namespace the key
+  - Value to be stored `v`
+
+  and stores `v` against key constructed using `s` and `t` in DB. Returns nil.
+
+  Optionally, you can pass in `:t` and `:valfn` as follows:
+
+  - `:t` will be used along with `s` to construct key. `t` should of type `Tuple` class.
+  If `t` is not passed, empty `Tuple` will be used.
```
Comment (vedang) [inline src/clj_fdb/core.clj]: Make `t` a compulsory argument. If the user wants to set a value against the empty tuple, he can pass the tuple in himself.

File: src/clj_fdb/core.clj [hunk 2]
```diff
@@ -120,3 +123,110 @@
...
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will be used to namespace the key
+
+  and returns byte-array `v` against key constructed using `s` and `t`.
+
+  Optionally, you can pass in `:t` and `:valfn` as follows:
+
+  - `:t` will be used along with `s` to construct key. `t` should of type `Tuple` class.
+  If `t` is not passed, empty `Tuple` will be used.
```
Comment (vedang) [inline src/clj_fdb/core.clj]: Same comment here as above

File: src/clj_fdb/core.clj [hunk 3]
```diff
@@ -120,3 +123,110 @@
...
+            valfn identity}}]
+   {:pre [(instance? Tuple t)]}
+   (get tc t
+        :keyfn #(fsubspace/pack s %)
+        :valfn valfn)))
+
+(defn clear-subspaced-key
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will used to namespace the range
```
Comment (vedang) [inline src/clj_fdb/core.clj]: Fix docstring: the subspace is used to namespace key (not range)

File: src/clj_fdb/core.clj [hunk 4]
```diff
@@ -120,3 +123,110 @@
...
+
+  and clears the Subspaced key from db. Returns nil.
+
+  Optionally, you can pass in `:t` as follows:
+
+  - `:t` will be used along with `s` to construct namespaced key. `t` should be of type `Tuple`.
+  If `t` is not passed, empty `Tuple` will be used."
```
Comment (vedang) [inline src/clj_fdb/core.clj]: Same comment here, make the tuple compulsory

File: src/clj_fdb/core.clj [hunk 5]
```diff
@@ -120,3 +123,110 @@
...
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will used to namespace the range
+
+  and returns a map of key/value pairs (byte-array->byte-array).
+
+  Optionally, you can pass in `:t`, `:keyfn` and `:valfn` as follows:
+
+  - `:t` will be used along with `s` to construct namespaced key. `t` should be of type `Tuple`.
+  If `t` is not passed, empty `Tuple` will be used.
```
Comment (vedang) [inline src/clj_fdb/core.clj]: Same as above

File: src/clj_fdb/core.clj [hunk 6]
```diff
@@ -120,3 +123,110 @@
...
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will used to namespace the range
+
+  and clears the range from db. Returns nil.
+
+  Optionally, you can pass in `:t` as follows:
+
+  - `:t` will be used along with `s` to construct namespaced key. `t` should be of type `Tuple`.
+  If `t` is not passed, empty `Tuple` will be used."
```
Comment (vedang) [inline src/clj_fdb/core.clj]: In this case, it makes sense to take `t` optionally. However, in the case that `t` is not provided, empty tuple should not be used (instead just call the single-arg version of `fsubspace/range`).

**Comment by pardeep-singh** [comment]
@vedang Made changes as per your suggestions. Please review the changes.

**Comment by vedang** [comment]
LGTM, merged

**Pull request merged**